### ISSUE TITLE
Re: BCM943602CS setup

### ISSUE THREAD

**Issue by ishtiaq156** [Errrneist/Hackintosh-Thinkpad-X1-Extreme#58]
_Originally posted by @ishtiaq156 in https://github.com/Errrneist/Hackintosh-Thinkpad-X1-Extreme/issues/11#issuecomment-568557049_

> Finally I can install the Mac successfully. Both Windows and Mac bluetooth drivers are working. Thanks so much guys :D

Hey @anht3889 , can you describe what you did to make bluetooth work in both Windows and macOS?
I am in the same situation as you were before:
- Bought **BCM943602CS** from TaoBao (Superbuy as proxy to get it delivered to USA)
- Replaced the built-in WiFi with 2CS and attached the two built-in antennas (or those wires - not sure what they are called - one is grey and other is black) to **J0** (**black**) and **J1** (**grey**).
- The 2CS card already came with a cable attached to **J2**. The end is just an antenna (with an adhesive to keep the cable in place?) which I left it as is.
- Also, there's one more cable - which has **blue-ends** with a white strip. I looked at photos which @Errrneist posted in the repository and attached the other end to the **slot**.

For macOS:
- WiFi worked as fine. No bluetooth.

For Windows:
- After installing drivers and rebooting, Wifi worked as fine. No bluetooth

I'm on latest EFI release which is **3.6**, and macOS **10.15.2**, if it matters.

**TL;DR**:
I attached cables to **2CS** card as follows (attached photo):
- **J0** to **black** cable
- **J1** to **grey** cable
- **J2** (left it as is - comes attached with an antenna)

To both @Errrneist and @anht3889 and (everyone else if they know what this is):
- Did I do it _right_?
- @anht3889, can you lay out the _steps_ you did to solve the BT issues please?

Also, thanks for the awesome guide, @Errrneist. I have been on my macOS for than 2 weeks now and it runs smooth as hell.

![IMG_0066](https://user-images.githubusercontent.com/25985759/71375705-28011f00-258d-11ea-9ffe-f2312bd41cc6.JPG)

_Originally posted by @ishtiaq156 in https://github.com/Errrneist/Hackintosh-Thinkpad-X1-Extreme/issues/11#issuecomment-568557049_

**Comment by wu-hongjun** [comment]
> _Originally posted by @ishtiaq156 in [#11 (comment)](https://github.com/Errrneist/Hackintosh-Thinkpad-X1-Extreme/issues/11#issuecomment-568557049)_
> 
> > Finally I can install the Mac successfully. Both Windows and Mac bluetooth drivers are working. Thanks so much guys :D
> 
> Hey @anht3889 , can you describe what you did to make bluetooth work in both Windows and macOS?
> I am in the same situation as you were before:
> 
> * Bought **BCM943602CS** from TaoBao (Superbuy as proxy to get it delivered to USA)
> * Replaced the built-in WiFi with 2CS and attached the two built-in antennas (or those wires - not sure what they are called - one is grey and other is black) to **J0** (**black**) and **J1** (**grey**).
> * The 2CS card already came with a cable attached to **J2**. The end is just an antenna (with an adhesive to keep the cable in place?) which I left it as is.
> * Also, there's one more cable - which has **blue-ends** with a white strip. I looked at photos which @Errrneist posted in the repository and attached the other end to the **slot**.
> 
> For macOS:
> 
> * WiFi worked as fine. No bluetooth.
> 
> For Windows:
> 
> * After installing drivers and rebooting, Wifi worked as fine. No bluetooth
> 
> I'm on latest EFI release which is **3.6**, and macOS **10.15.2**, if it matters.
> 
> **TL;DR**:
> I attached cables to **2CS** card as follows (attached photo):
> 
> * **J0** to **black** cable
> * **J1** to **grey** cable
> * **J2** (left it as is - comes attached with an antenna)
> 
> To both @Errrneist and @anht3889 and (everyone else if they know what this is):
> 
> * Did I do it _right_?
> * @anht3889, can you lay out the _steps_ you did to solve the BT issues please?
> 
> Also, thanks for the awesome guide, @Errrneist. I have been on my macOS for than 2 weeks now and it runs smooth as hell.
> 
> ![IMG_0066](https://user-images.githubusercontent.com/25985759/71375705-28011f00-258d-11ea-9ffe-f2312bd41cc6.JPG)
> 
> _Originally posted by @ishtiaq156 in [#11 (comment)](https://github.com/Errrneist/Hackintosh-Thinkpad-X1-Extreme/issues/11#issuecomment-568557049)_

For Bluetooth in Windows, you’ll need to install the driver I mentioned in the reader.MD, I think it is in the pre-install module. So far everything looks right...if it is not showing up Hummmm try check the connections of the two ends of the ribbon cables.

**Comment by ishtiaq156** [comment]
> For Bluetooth in Windows, you’ll need to install the driver I mentioned in the reader.MD

Yes. I installed the driver as mentioned.



> if it is not showing up Hummmm try check the connections of the two ends of the ribbon cables

This. I think this is what the issue is for me. The slot which open/closes is a little lose and it doesn't just fit in. No click sounds or any feedback when I close the lid for that ribbon cable. Is it the same for you?

**Comment by wu-hongjun** [comment]
> > For Bluetooth in Windows, you’ll need to install the driver I mentioned in the reader.MD
> 
> Yes. I installed the driver as mentioned.
> 
> > if it is not showing up Hummmm try check the connections of the two ends of the ribbon cables
> 
> This. I think this is what the issue is for me. The slot which open/closes is a little lose and it doesn't just fit in. No click sounds or any feedback when I close the lid for that ribbon cable. Is it the same for you?

The ribbon cable should have a nice feedback...the two ends should provide the same experience when you insert. If not, you probably have a faulty board and should ask for a replacement...idk

**Comment by ishtiaq156** [comment]
Thank you for the response. I'll close this issue now.

**Comment by wu-hongjun** [comment]
> Thank you for the response. I'll close this issue now.

If it is not showing up in Windows and Mac the only thing I can think of is hardware issue.

**Comment by ishtiaq156** [comment]
Indeed. It was that ribbon cable retainer thing. I opened up the case and tighten it up a bit again and that’s about it. Got bluetooth as well in both Windows and macOS. No more dongles now. Thanks!

**Comment by wu-hongjun** [comment]
> Indeed. It was that ribbon cable retainer thing. I opened up the case and tighten it up a bit again and that’s about it. Got bluetooth as well in both Windows and macOS. No more dongles now. Thanks!

K cool!
